libmscore/measure.cpp

```diff
@@ -250,8 +250,14 @@ Measure::Measure(const Measure& m)
 
 void Measure::layoutStaffLines()
       {
-      for (MStaff* ms : _mstaves)
-            ms->lines()->layout();
+      int staffIdx = 0;
+      for (MStaff* ms : _mstaves) { 
+            if (score()->staff(staffIdx)->cutaway() && isCourtesyClef(staffIdx)) 
+                  ms->lines()->layoutPartial(width(), 4.0 ,false);    // draw a shortened staff (4.0 sp wide) only for the antecipated Clef.
+            else
+                ms->lines()->layout();
+            staffIdx += 1;
+            }  
       }
 
```

Comment:
I wrote this function lines()->layoutPartial() (maybe need a better name) that draws only a certain width of the lines and aligns them to the left or right of a measure. In this specific case I needed just enough lines width to fit any Clef.

On a related topic, I have been looking for a way to overcome MuseScore's limitation of having a staff change in the middle of a measure. I believe this function might give a clue of how to do that. The code would:
- Loop thru the staffTypes in the measure
- A function should calculate the ms->lines() to be added. 
- Example: if there is a staff change from 5 to 1 line in the middle of the measure:, 6 lines will be added to ms->lines: 5 from the start of the measure to the tick of the stchange, and one from there to the end of the measure width. There would be no limit